File: Arrays/zeros_to_tail.py
#!/usr/bin/env python3

def zeros_to_tail(arr):
    zeroes = 0
    end_of_nonzero_numbers = 0
    for num in arr:
        if num == 0:
            zeroes += 1
        else:
            arr[end_of_nonzero_numbers] = num
            end_of_nonzero_numbers += 1
    for idx in range(end_of_nonzero_numbers, len(arr)):
        arr[idx] = 0
    return arr



test = [1, 0, 2, 0, 3, 0, 0, 0, 4, 5, 0, 69, 0]

print(zeros_to_tail(test))

File: Arrays/zeros_to_tail.rb
def zeros_to_tail(arr)
    zereos=0
    end_of_nonzero_numbers=0
    arr.each do |item|
        if item==0
            zereos+=1
        else
            arr[end_of_nonzero_numbers] = item
            end_of_nonzero_numbers += 1
        end
    end
    
    while end_of_nonzero_numbers<arr.length do
        arr[end_of_nonzero_numbers]=0
        end_of_nonzero_numbers+=1
    end
return arr
end

test_array = [1, 0, 2, 0, 0, 0, 0, 0, 4, 5, 0, 69, 0]

puts zeros_to_tail(test_array)